api/dependencies.go

```diff
@@ -36,9 +36,7 @@ type VM interface {
 		verifySig bool,
 		txs []*chain.Transaction,
 	) (errs []error)
-	// LastAcceptedBlock provides the most recent block that the VM has accepted.
-	// The value returned is guaranteed to be non-nil.
-	LastAcceptedBlock() *chain.StatefulBlock
+	LastAcceptedBlockResult() *chain.ExecutedBlock
```

Comment:
nit: Feel like the rename isn't needed, since the result is technically a field encoded in the block type

Reply:
Open to suggestion to eliminate the need for any of them https://github.com/ava-labs/hypersdk/pull/1744#discussion_r1834517278